Compilation Warnings

just one warning:

src/scram.c: In function ‘build_server_final_message’:
src/scram.c:977:34: warning: ‘%s’ directive output between 0 and 2147483646 bytes may cause result to exceed ‘INT_MAX’ [-Wformat-truncation=]
  977 |         snprintf(result, len, "v=%s", server_signature);

GCC 12.x
Fedora 37 64bits

dunno where to fix the code...

--- a/src/scram.c
+++ b/src/scram.c
@@ -971,6 +971,16 @@ char *build_server_final_message(ScramState *scram_state)
 		goto failed;
 
 	len = 2 + strlen(server_signature) + 1;
+
+	/*
+	 * Avoid compiler warning at snprintf() below because len
+	 * could in theory overflow snprintf() result.  If this
+	 * happened in practice, it would surely be some crazy
+	 * corruption, so treat it as an error.
+	 */
+	if (len >= INT_MAX)
+		goto failed;
+
 	result = malloc(len);
 	if (!result)
 		goto failed;

